src/Feature/Layout/Feature_Layout.rei

```diff
@@ -0,0 +1,53 @@
+type direction =
+  | Up
+  | Left
+  | Down
+  | Right;
+
+// definition only used for tests
+type t('content) =
+  | Split([ | `Horizontal | `Vertical], list(t('content)))
+  | Window({
+      weight: float,
+      content: 'content,
+    })
+  | Empty;
+
+[@deriving show]
+type window('content) = {
+  content: 'content,
+  x: int,
+  y: int,
+  width: int,
+  height: int,
+};
+
+module Internal: {
+  let move:
+    ('content, int, int, list(window('content))) => option('content); // only used for tests
+};
+
+let initial: t('content);
+
+let windows: t('content) => list('content);
+let addWindow:
+  (
+    ~target: option('content)=?,
+    ~position: [ | `Before | `After],
+    [ | `Horizontal | `Vertical],
```

Comment:
I wanted to unify all these arguments into one that accurately models the problem instead of ignoring parts of the input in certain conditions. It could be a single argument with ``[`Above('content) | `Below('content) | `LeftOf('content) | 'RIghtOf(content) | `Leftmost | `Rightmost]`` but the way it's currently used makes that a bit awkward in some places.